python/foglamp/services/core/api/filters.py

```diff
@@ -547,3 +437,65 @@ def new_items(lst1, lst2):
     # Removed key from configuration cache
     config_mgr = ConfigurationManager(storage)
     config_mgr._cacheManager.remove(key)
+
+
+def _diff(lst1: List[str], lst2: List[str]) -> List[str]:
+    return [v for v in lst2 if v not in lst1]
+
+
+def _delete_keys_from_dict(dict_del: Dict, lst_keys: List[str], deleted_values: Dict = {}, parent: str = None) -> Tuple[
+    Dict, Dict]:
+    """ Delete keys from the dict and add deleted key=value pairs to deleted_values dict"""
+    for k in lst_keys:
+        try:
+            if parent is not None:
+                deleted_values.update({parent: dict_del[k]})
+            del dict_del[k]
+        except KeyError:
+            pass
+    for k, v in dict_del.items():
+        if isinstance(v, dict):
+            parent = k
+            _delete_keys_from_dict(v, lst_keys, deleted_values, parent)
+    return dict_del, deleted_values
+
+
+async def _delete_child_filters(storage: StorageClientAsync, cf_mgr: ConfigurationManager, service_name: str,
+                                   new_list: List[str], old_list: List[str] = []) -> None:
+    # Difference between pipeline and value from storage lists and then delete relationship as per diff
+    delete_children = _diff(new_list, old_list)
+    for l in delete_children:
+        try:
+            filter_child_category_name = "{}_{}".format(service_name, l)
+            await cf_mgr.delete_child_category(service_name, filter_child_category_name)
+        except:
+            pass
+        payload = PayloadBuilder().WHERE(['key', '=', "{}_{}".format(service_name, l)]).payload()
+        await storage.delete_from_tbl("configuration", payload)
+        payload = PayloadBuilder().WHERE(['name', '=', l]).AND_WHERE(['user', '=', service_name]).payload()
+        await storage.delete_from_tbl("filter_users", payload)
+
+
+async def _add_child_filters(storage: StorageClientAsync, cf_mgr: ConfigurationManager, service_name: str,
+                               filter_list: List[str], old_list: List[str] = []) -> None:
+    # Create children categories. Since create_category() does not expect "value" key to be
+    # present in the payload, we need to remove all "value" keys BUT need to add back these
+    # "value" keys to the new configuration.
+    for filter_name in filter_list:
+        filter_config = await cf_mgr.get_category_all_items(category_name=filter_name)
+        filter_desc = "Configuration of {} filter for user {}".format(service_name, filter_name)
```

Comment:
swap `service_name, filter_name` args

Reply:
I think `service_` should not be used, as filters can be applied with north instances too.